src/query/api/v1/handler/prom/common.go

```diff
@@ -53,7 +53,7 @@ const (
 
 type errorType string
 
-type queryData struct {
+type QueryData struct {
```

Comment:
To me, it doesn't look like `queryData` needs to be public, at least in the scope of this PR (ditto for `Respond` and `RespondError`). Am I missing something?

Reply:
Since we can now override default handlers, we might need these for such use-cases.

Reply:
I think we should take a bit more care when exposing things. For example, there seems to be a duplicate of this struct in `prometheus/response.go`, which doesn't surface types from prometheus repository. Maybe we should pick one of those to be public, and not both? 

https://github.com/m3db/m3/blob/2bcd6fd24dee8b6d81f89a201745de1933aca834/src/query/api/v1/handler/prometheus/response.go#L40-L69

Similar question with `Respond` and `RespondError` – are these the methods we want to have exposed? Aren't there better candidates?

Reply:
`response` is not made public in this PR so no changes there.
As for `Respond` and `RespondError`, I don't think exposing them would make any harm. They do not belong to any class and could be reused by custom handlers without  a need to copy the same code (as we are currently doing in our private codebase).

Reply:
nit: Need to add comment to exported types `// QueryData is ...`.